microsetta_private_api/repo/perk_fulfillment_repo.py

```diff
@@ -0,0 +1,591 @@
+from microsetta_private_api.config_manager import SERVER_CONFIG
+
+
+class PerkFulfillmentRepo(BaseRepo):
+    def __init__(self, transaction):
+        super().__init__(transaction)
+
+    @staticmethod
+    def _row_to_subscription(r):
+        return Subscription.from_dict(r)
+
+    def find_perks_without_fulfillment_details(self):
+        """If a perk isn't represented in our system at all, the transaction
+        won't be imported. However, we need to have fulfillment details to
+        act on a perk being selected. This finds any perks for which we
+        haven't established fulfillment details and returns content for an
+        alert email.
+        """
+        with self._transaction.dict_cursor() as cur:
+            cur.execute(
+                "SELECT transaction_id, perk_id "
+                "FROM campaign.fundrazr_transaction_perk "
+                "WHERE perk_id NOT IN ("
+                "    SELECT DISTINCT(perk_id) "
+                "    FROM campaign.fundrazr_perk_fulfillment_details"
+                ")"
+            )
+            rows = cur.fetchall()
+            if rows is None:
+                return None
+            else:
+                return [dict(r) for r in rows]
+
+    def process_pending_fulfillments(self):
+        """Find all purchased perks that have yet to be processed
+        and fulfill them"""
+        error_report = []
+
+        with self._transaction.dict_cursor() as cur:
+            cur.execute(
+                "SELECT ftp.id ftp_id, ftp.transaction_id, ftp.perk_id, "
+                "ftp.quantity, ft.payer_email, fpfd.ffq_quantity, "
+                "fpfd.kit_quantity, fpfd.dak_article_code, "
+                "fpfd.fulfillment_spacing_number, "
+                "fpfd.fulfillment_spacing_unit, iu.first_name, iu.last_name, "
+                "iu.phone, iu.address_1, iu.address_2, iu.city, iu.state, "
+                "iu.postal_code, iu.country, iu.campaign_id "
+                "FROM campaign.fundrazr_transaction_perk ftp "
+                "INNER JOIN campaign.transaction ft "
+                "ON ftp.transaction_id = ft.id "
+                "INNER JOIN campaign.fundrazr_perk_fulfillment_details fpfd "
+                "ON ftp.perk_id = fpfd.perk_id "
+                "INNER JOIN campaign.interested_users iu "
+                "ON ft.interested_user_id = iu.interested_user_id "
+                "AND iu.address_valid = true "
+                "WHERE ftp.processed = false"
+            )
+            rows = cur.fetchall()
+
+            for row in rows:
+                if (row['ffq_quantity'] > 1 or row['kit_quantity'] > 1) and \
+                        row['fulfillment_spacing_number'] > 0:
+                    subscription_id = \
+                        self._create_subscription(row['payer_email'],
+                                                  row['transaction_id'],
+                                                  row['ftp_id'])
```

Comment:
```suggestion
                if has_subscription(row): 
                    subscription_id = \
                        self._create_subscription(row['payer_email'],
                                                  row['transaction_id'],
                                                  row['ftp_id'])
                else:
                    subscription_id = None 
```

...and add a method elsewhere:

```python
def has_subscription(d):
    return (d['ffq_quantity'] > 1 or d['kit_quantity'] > 1) and \
           (d['fulfillment_spacing_number'] > 0)